File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/web/webapp/mappers/HospitalDTOsMapper.java
package io.zenwave360.example.clinicaltool.modules.web.webapp.mappers;

import io.zenwave360.example.clinicaltool.common.mappers.*;
import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.*;
import io.zenwave360.example.clinicaltool.modules.clinical.core.inbound.dtos.*;
import io.zenwave360.example.clinicaltool.modules.web.webapp.dtos.*;
import java.math.*;
import java.time.*;
import java.util.*;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

@Mapper(uses = BaseMapper.class)
public interface HospitalDTOsMapper {

    HospitalDTOsMapper INSTANCE = Mappers.getMapper(HospitalDTOsMapper.class);

    // request mappings
    Doctor asDoctor(DoctorDTO dto);

    Hospital asHospital(HospitalDTO dto);

    // response mappings

    HospitalDTO asHospitalDTO(Hospital entity);

    DoctorDTO asDoctorDTO(Doctor entity);

    List<DoctorDTO> asDoctorDTOList(List<Doctor> entityList);

    List<HospitalDTO> asHospitalDTOList(List<Hospital> entityList);

    List<PatientHospitalDTO> asPatientHospitalDTOList(List<PatientHospital> entityList);
}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/web/webapp/mappers/HospitalDTOsMapper.kt
package io.zenwave360.example.clinicaltool.modules.web.webapp.mappers

import io.zenwave360.example.clinicaltool.common.mappers.*
import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.*
import io.zenwave360.example.clinicaltool.modules.clinical.core.inbound.dtos.*
import io.zenwave360.example.clinicaltool.modules.web.webapp.dtos.*

import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.factory.Mappers
import java.math.*
import java.time.*
import java.util.*
import org.springframework.data.domain.Page

@Mapper(uses = [BaseMapper::class])
interface HospitalDTOsMapper {

    companion object {
        val INSTANCE: HospitalDTOsMapper = Mappers.getMapper(HospitalDTOsMapper::class.java)
    }

    // request mappings
    fun asDoctor(dto: DoctorDTO): Doctor
    fun asHospital(dto: HospitalDTO): Hospital

    // response mappings
    
    fun asPatientHospitalDTO(entity: PatientHospital): PatientHospitalDTO
    
    
    fun asHospitalDTO(entity: Hospital): HospitalDTO
    
    
    fun asDoctorDTO(entity: Doctor): DoctorDTO
    
    
    fun asDoctorDTOList(entityList: List<Doctor>): List<DoctorDTO>
    
    fun asHospitalDTOList(entityList: List<Hospital>): List<HospitalDTO>
    
    fun asPatientHospitalDTOList(entityList: List<PatientHospital>): List<PatientHospitalDTO>

}
